tests/test_model_api.py

```diff
@@ -0,0 +1,101 @@
+import os
+import pytest
+import sympy as sp
+
+from cellmlmanip import parser
+
+
+OXMETA = "https://chaste.comlab.ox.ac.uk/cellml/ns/oxford-metadata#"
+
+
+class TestModelAPI(object):
+    """
+    Tests various parts of the Model API.
+    """
+
+    @pytest.fixture(scope="class")
+    def model(self):
+        path = os.path.join(
+            os.path.dirname(__file__), 'cellml_files', 'hodgkin_huxley_squid_axon_model_1952_modified.cellml')
+        return parser.Parser(path).parse()
+
+    def test_add_unit(self, model):
+        """ Tests the add_unit method. """
+
+        # TODO: Add unit tests for normal usage
+
+        # Base units can't have attributes
+        with pytest.raises(ValueError, match='can not be defined with unit attributes'):
```

Comment:
Should also test you get an error if the unit already exists?

Reply:
Someone should, but not me :D
Only added a test for this because I changed an `assert` into a `throw`. All other fixes are out of scope of this PR, I think